bagua/torch_api/algorithms/decentralized.py

```diff
@@ -53,23 +55,40 @@ def hook(parameter_name, parameter):
 
     def init_post_backward_hook(self, bagua_module: BaguaModule):
         def hook():
-            bagua_module._bagua_backend.wait_pending_comm_ops()
-            torch.cuda.synchronize()
-            bagua_module._bagua_backend.execute_post_backward_comm_ops()
-            bagua_module._bagua_backend.wait_pending_post_backward_comm_ops()
+            if self.step % self.communication_interval == 0:
+                bagua_module._bagua_backend.wait_pending_comm_ops()
+
+                intra_comm = bagua_module._bagua_backend.intranode_communicator
+                inter_comm = bagua_module._bagua_backend.internode_communicator
+
+                for bucket in bagua_module.bagua_buckets:
+                    if not self.hierarchical or (inter_comm is not None):
+                        bucket.backend_tensor.copy_(bucket._peer_weight)
+
+                    if self.hierarchical:
+                        broadcast(bucket.backend_tensor, 0, intra_comm)
+
+            self.step += 1
 
         return hook
 
+    def _init_states(self, bucket: BaguaBucket):
+        weight_tensor = bucket.flattened_tensor()
+        bucket._peer_weight = weight_tensor.to_bagua_tensor("peer_weight")
+
     def init_operations(
         self,
         bagua_module: BaguaModule,
         bucket: BaguaBucket,
     ):
+        self._init_states(bucket)
+        torch.cuda.synchronize()
         bucket.clear_ops()
         bucket.append_decentralized_synchronous_op(
             hierarchical=self.hierarchical,
             peer_selection_mode=self.peer_selection_mode,
             communication_interval=self.communication_interval,
```

Comment:
remove communication op, only mark tensor ready when `bagua_train_step_counter` % comm_interval == 0

Reply:
> remove communication interval, only mark tensor ready when `bagua_train_step_counter` % comm_interval == 0

fixed, add a new method `_bagua_not_skip_this_step` in `BaguaModule`